### PR TITLE
Vocabulary page

### PR DESCRIPTION
**Pull request by Mark169Z**
This is a vocabulary page. 

Not finished -> 
Time Sort
Final Design
Some components

### CODE REVIEW

File: client/src/components/SearchbarTopicProblem.jsx
Comment (Chatchaponz) [inline client/src/components/SearchbarTopicProblem.jsx:1]: Remove empty file na

File: client/src/components/SearchbarWord.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            onSearch(query);
+        }
+    };
+    const handleKeyPress = (e) => {
+        if (e.key === 'Enter') {
+            handleSearch();
+        }
+    };
+    const navigate = useNavigate() // for React Router v6
+    const location = useLocation() // Get the current location
```
Comment (Chatchaponz) [inline client/src/components/SearchbarWord.jsx:22]: How these consts being use?

File: client/src/pages/vocabulary/VocabularyPage.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            setData(sorted);
+        } else if (order === 'DSC') {
+            const sorted = [...data].sort((a, b) =>
+                b.d1.localeCompare(a.d1, 'th')
+            );
+            setData(sorted);
+        }
+    }
+
+    // Yang mai dai tum ->
```
Comment (Chatchaponz) [inline client/src/pages/vocabulary/VocabularyPage.jsx:92]: Got it lol

File: client/src/pages/vocabulary/VocabularyPage.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        if (order === 'ASC') {
+            const sorted = [...data].sort((a, b) =>
+                a.d1.localeCompare(b.d1, 'th')
+            );
+            setData(sorted);
+        } else if (order === 'DSC') {
+            const sorted = [...data].sort((a, b) =>
+                b.d1.localeCompare(a.d1, 'th')
+            );
+            setData(sorted);
```
Comment (Chatchaponz) [inline client/src/pages/vocabulary/VocabularyPage.jsx:88]: I think this can be simplify, but it up to you though

**Comment by Chatchaponz** [comment]
Dont forget to resolve conflict na

File: client/src/components/SearchbarTopicProblem.jsx
Comment (Mark169Z) [reply client/src/components/SearchbarTopicProblem.jsx:1]: Resolved ✅

File: client/src/pages/vocabulary/VocabularyPage.jsx (hunk 3 above)
Comment (Mark169Z) [reply client/src/pages/vocabulary/VocabularyPage.jsx]: Simplified ✅

File: client/src/components/SearchbarWord.jsx (hunk 1 above)
Comment (Mark169Z) [reply client/src/components/SearchbarWord.jsx]: Finished

File: client/src/pages/vocabulary/VocabularyPage.jsx (hunk 2 above)
Comment (Mark169Z) [reply client/src/pages/vocabulary/VocabularyPage.jsx]: Finished

File: client/src/components/Searchbar.jsx [hunk 4]
```diff
@@ -37,7 +36,7 @@ function SearchBar({
               </Combobox.Option>
             ))}
           </Combobox.Options>
-        )}
+        )} */}
```
Comment (Chatchaponz) [inline client/src/components/Searchbar.jsx:39]: Do you need this? if not pls remove krub

File: client/src/general/Router.jsx [hunk 5]
```diff
@@ -36,7 +37,17 @@ export const Router = () => (
...
+          <Navbaruser/>
+          <Vocabulary />
+        </Layout>
+      }
+    /> 
   </Routes>
+  </>
 );
 
 // <Route path="" element="">
```
Comment (Chatchaponz) [inline client/src/general/Router.jsx:53]: Dont forget to remove unused comment na

File: client/src/pages/vocabulary/VocabularyPage.jsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+import MyDropdown from "./_components/Dropdown.jsx";
+import axios from 'axios';
+
+export const Vocabulary = () => {
+    const [searchResults, setSearchResults] = useState([]);
+    const [activeRowIndex, setActiveRowIndex] = useState(null);
+    const [type, setType] = useState("ASC");
+    const [page, setPage] = useState(1);
+    const [category, setCategory] = useState(null);
+    // const [interpreter, setInterpreter] = useState(null);
```
Comment (Chatchaponz) [inline client/src/pages/vocabulary/VocabularyPage.jsx:16]: rm

File: client/src/pages/vocabulary/VocabularyPage.jsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+                    find: searchbar || '',
+                    category: category || '',
+                    parts_of_speech: wordtype || '',
+                    page: pageNum
+                }
+            });
+    
+            let results = Array.isArray(response.data.suggestions) ? response.data.suggestions : [];
+    
+            console.log("test")
```
Comment (Chatchaponz) [inline client/src/pages/vocabulary/VocabularyPage.jsx:64]: Don't forget to rm this na if not use

File: server/controllers/ticketController.js [hunk 8]
```diff
@@ -122,14 +85,12 @@ export const updateTicket = async (req, res) => {
 export const deleteTicket = async (req, res) => {
   try {
     const { id } = req.query;
-
+    // console.log(id);
```
Comment (Chatchaponz) [inline server/controllers/ticketController.js:88]: rm

File: client/src/pages/vocabulary/VocabularyPage.jsx (hunk 6 above)
Comment (Mark169Z) [reply client/src/pages/vocabulary/VocabularyPage.jsx:16]: Removed

File: client/src/components/Searchbar.jsx (hunk 4 above)
Comment (Mark169Z) [reply client/src/components/Searchbar.jsx:39]: Deleted

File: client/src/general/Router.jsx (hunk 5 above)
Comment (Mark169Z) [reply client/src/general/Router.jsx:53]: Removed

File: client/src/pages/vocabulary/VocabularyPage.jsx (hunk 7 above)
Comment (Mark169Z) [reply client/src/pages/vocabulary/VocabularyPage.jsx:64]: rm

File: client/src/pages/vocabulary/VocabularyPage.jsx [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                SignBloom3D
+            </div>
+
+            <div className="flex flex-col justify-center items-center mt-14 lg:w-[57rem] w-3/4">
+            <Searchbar setSearchbar={setSearchbar} />
+
+                <div className="flex flex-row justify-between w-full mt-4 gap-4">
+                    <FilterCategory setCategory={setCategory} />
+                    <FilterWordtype setWordtype={setWordtype} />
+                    {/* <FilterInterpreter setInterpreter={setInterpreter} /> */}
```
[the remark this answers is not in the document]
Comment (Mark169Z) [reply client/src/pages/vocabulary/VocabularyPage.jsx:127]: rm

File: server/controllers/ticketController.js (hunk 8 above)
Comment (Mark169Z) [reply server/controllers/ticketController.js:88]: rm

**Pull request merged**